### PR TITLE
New Unit: Rocket Launcher

### PR DESCRIPTION
**Pull request by YikaiZheng**
Created a new unit: rocket launcher together with the corresponding bullet and particle
The description of the unit is in rocket_launcher.cpp
The class name is "Cart", in order to be simple, but it can be changed to "RocketLauncher" if necessary.

### CODE REVIEW

**Comment by LazyJazz** [comment]
Could u pls upload some screenshots. It could help us understand what you have done.

**Comment by YikaiZheng** [comment]
https://user-images.githubusercontent.com/113595341/211175758-7508d676-d027-4816-8fa8-e0dd677719c0.mp4

**Comment by YikaiZheng** [comment]
The target sign is where I clicked. You see that the cart backed off a little after firing. The first two shoots hit the target, but the third failed due to randomness (The long distance caused low precision)
Hope u guys enjoy it~

**Review by LazyJazz — CHANGES_REQUESTED** [review]
LGTM! Just a few nits.

**Comment by YikaiZheng** [comment]
I've addressed the changes. Hope this can be merged~

File: src/battle_game/core/units/rocket_launcher.cpp [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    if (input_data.key_down[GLFW_KEY_A]) {
+      rotation_offset += 1.0f;
+    }
+    if (input_data.key_down[GLFW_KEY_D]) {
+      rotation_offset -= 1.0f;
+    }
+    rotation_offset *= kSecondPerTick * rotate_angular_speed * GetSpeedScale();
+    game_core_->PushEventRotateUnit(id_, rotation_ + rotation_offset);
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/rocket_launcher.cpp:105]: Inherit from unit::Tank if you are using the same function.

File: src/battle_game/core/units/rocket_launcher.cpp [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+void Cart::TurretRotate() {
+  auto player = game_core_->GetPlayer(player_id_);
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    }
+    turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+  }
```
Comment (LazyJazz) [inline src/battle_game/core/units/rocket_launcher.cpp:115]: These too.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/rocket_launcher.cpp (hunk 1 above)
Comment (YikaiZheng) [reply src/battle_game/core/units/rocket_launcher.cpp:105]: But there's a different conditional branch here, so will it be kind of tricky?

**Comment by YikaiZheng** [comment]
I inherited from Tank and got rid of TurrentRotate, but CartMove is slightly different from TankMove in that there's backforce after firing, so it may not save a lot of work by making use of TankMove, and I kept it as it was.

**Review by LazyJazz — DISMISSED** [review]
LGTM! Call me in wechat if you want to merge this pr into the main branch.

**Pull request closed without merging**